design/fast_finality.md

```diff
@@ -0,0 +1,273 @@
+# Fast finality design
+
+## Scope
+
+The introduction of a single sequencer to improve cost-per-transaction during the bootstrapping phase, as 
+described in the [Bootstrapping Strategy design doc](./Bootstrapping_strategy.md).
+
+## Glossary
+
+* *Light batch*: A set of transactions produced by the designated sequencer, considered to be soft-finalised in that 
+  order
+
+## Requirements
+
+* Finality
+  * Transaction *soft* finality (finality guaranteed by the sequencer) is achieved in under one second
+  * There is eventual transaction *hard* finality (finality guaranteed by the L1)
+  * The sequencer is incentivised to achieve hard finality on an agreed cadence
+  * The sequencer is incentivised to hard-finalise transactions in the same order they are soft-finalised
+  * The sequencer is not able to "rewrite history" (or is strongly disincentivised from doing so), even for soft-final
+    transactions
+  * Soft finality does not break the mechanism used by some smart contracts (e.g. SushiSwap) of using `block.number` as 
+    a rudimentary clock
+* Censorship resistance
+  * End-users have a mechanism to force the sequencer to include their transactions (possibly at higher cost and slower 
+    finality)
+  * The sequencer distributes all soft-finalised transactions to all nodes
+  * Nodes can easily prove whether a given soft-finalised transaction was hard-finalised, and in the correct order
```

Comment:
I would group this requirement with the others related to how the sequencer is forced to behave